investment_tracking/views.py

```diff
@@ -18,14 +20,43 @@ class InvestmentTrackingSaveView(APIView):
     def post(self, request, startup_id):
         investor_profile = get_object_or_404(InvestorProfile, user=request.user)
         startup = get_object_or_404(StartUpProfile, id=startup_id)
+
         serializer = InvestmentTrackingSerializerCreate(data={'investor': investor_profile.id, 'startup': startup.id})
         if serializer.is_valid():
             try:
                 investment_tracking = InvestmentTracking(investor=investor_profile, startup=startup)
                 investment_tracking.save()
-                return Response({"message": "StartUp has been successfully saved."}, status = status.HTTP_201_CREATED)
+                return Response({"message": "StartUp has been successfully saved."}, status=status.HTTP_201_CREATED)
             except IntegrityError:
-                return Response({"message": "Startup is already saved."}, status = status.HTTP_400_BAD_REQUEST)
+                return Response({"message": "Startup is already saved."}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class InvestmentTrackingListView(generics.ListAPIView):
+    """
+    API view to get list all startups saved by the investor.
+    """
+    permission_classes = [IsAuthenticated]
+    queryset = InvestmentTracking.objects.all()
```

Comment:
Directly setting queryset = InvestmentTracking.objects.all() means that the base queryset could be loaded unnecessarily.
Set queryset to None initially, and only retrieve it in get_queryset, which also improves performance.
queryset = None  # defer queryset until filtered by get_queryset

Reply:
Done